doc/user/content/overview/key-concepts.md

```diff
@@ -190,20 +191,63 @@ encoded in Avro with the [Debezium diff envelope](/sql/create-sink#debezium-enve
+cluster with no replicas does no computation. For example, if you create an
+index on a cluster with no replicas, you cannot select from that index because
+there is no physical representation of the index to read from.
+
+Though clusters only represent the logic of which objects you want to bundle
+together, this impacts the performance characteristics once you provision
+cluster replicas. Each object in a cluster gets instantiated on every replica,
+meaning they are all in contention for the same physical resources. To achieve
+the performance you need, this might require setting up more than one cluster.
+
+### Deployment options
+
+When building your Materialize deployment, you can change its performance
+characteristics by...
+
+Action | Outcome
+-------|---------
+Adding clusters + decreasing dataflow density | Reduced contention among dataflows, decoupled dataflow availability
+Adding replicas to clusters | See [Cluster replica scaling](#cluster-replica-scaling)
+
+## Cluster replicas
+
+Where clusters represent the logical set of dataflows you want to maintain,
+cluster replicas are their physical counterparts. Cluster replicas are where
+Materialize actually creates and maintains dataflows.
+
+Each cluster replica is essentially a clone, constructing the same dataflows.
+Each cluster replica receives a copy of all data that comes in from sources its
+dataflows use, and uses the data to performs identical computations. This design
+provides Materialize with active replication, and so long as one replica is
+still reachable, the cluster continues making progress.
+
+This also means that all of a cluster's dataflows contend for the same resources
+on each replica. This might mean, for instance, that instead of placing many
+complex materialized views on the same cluster, you choose some other
+distribution, or you replace all replicas in a cluster with more powerful
+machines.
+
+### Deployment options
+
+Materialize is an active-replication-based system, which means you expect each
+cluster replica to have the same working set.
 
-A cluster is a set of compute resources that have been allocated to maintain
-indexes and sinks. The `materialized` process provides one local cluster named
-`default` that represents the compute resources of the local machine. In
-a forthcoming version of [Materialize Cloud](/cloud), you will be able to
-dynamically create and drop clusters to allocate and deallocate compute
-resources on demand.
+With this in mind, when building your Materialize deployment, you can change its
+performance characteristics by...
 
-Clusters are still under active development and are not yet meant for general
-use. They are described here because some non-experimental features (e.g.,
-[`SHOW INDEX`](/sql/show-indexes)) already make reference to clusters.
+Action | Outcome
+---------|---------
+Increase replicas' memory + CPU | Ability to maintain more dataflows or more complex dataflows
```

Comment:
```suggestion
Increase replicas' size | Ability to maintain more dataflows or more complex dataflows
```

Since user's can't directly increase a replica's memory + CPU.